### PR TITLE
DAOS-13292 build: Don't need UCX libraries … (#15016)

### PR DESCRIPTION
**Pull request by kjacque**
At build time any more, as of e01970d.

Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by kjacque** [comment]
These are build changes only.

**Review by jolivier23 — COMMENTED** [review]
Remove utils/build.config and other stuff ? Maybe I asked this on the master PR as well but don't recall

**Comment by daltonbohning** [comment]
> Remove utils/build.config and other stuff ? Maybe I asked this on the master PR as well but don't recall

It's a little misleading. It's just removing UCX from the RPM dependency, but source builds still require and build UCX

File: utils/rpms/daos.spec [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 Name:          daos
 Version:       2.6.2
-Release:       2%{?relval}%{?dist}
+Release:       3%{?relval}%{?dist}
```
Comment (daltonbohning) [inline utils/rpms/daos.spec:18]: Well, oops. https://github.com/daos-stack/daos/commit/c631702067a1528b66445e84a4e9da998d4cd043 did not bump the `Release` (sorry, I merged that). So this should be `4` to be inline with the changelog comments. I guess we'll just skip `3`. @brianjmurrell Does that sound reasonable? Or should we keep the `Release: 3` and update the changelogs to be duplicated with `3` as well?

Comment (brianjmurrell) [reply utils/rpms/daos.spec:18]: Yeah, skip 3 here but leave it in the changelog(s).  As usual, keep the `debian/changelog` and the `daos.spec` in parity though.

**Review by brianjmurrell — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Comment by kjacque** [comment]
There is a known NLT valgrind issue on the 2.6 branch that caused the failures. All other tests passed for this PR. Requesting forced landing.

**Pull request merged**